components/proxy/src/test/java/com/hotels/styx/startup/StyxServerComponentsTest.java

```diff
@@ -0,0 +1,133 @@
+/**
+ * Copyright (C) 2013-2018 Expedia Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.styx.startup;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.hotels.styx.Environment;
+import com.hotels.styx.StyxConfig;
+import com.hotels.styx.api.HttpInterceptor.Chain;
+import com.hotels.styx.api.configuration.Configuration;
+import com.hotels.styx.api.metrics.codahale.CodaHaleMetricRegistry;
+import com.hotels.styx.api.plugins.spi.Plugin;
+import com.hotels.styx.api.service.spi.StyxService;
+import com.hotels.styx.proxy.plugin.NamedPlugin;
+import com.hotels.styx.startup.StyxServerComponents.LoggingSetUp;
+import com.hotels.styx.support.api.BlockingObservables;
+import org.testng.annotations.Test;
+
+import java.util.List;
+import java.util.Map;
+
+import static com.hotels.styx.api.HttpRequest.Builder.get;
+import static com.hotels.styx.api.HttpResponse.Builder.response;
+import static com.hotels.styx.proxy.plugin.NamedPlugin.namedPlugin;
+import static com.hotels.styx.support.matchers.IsOptional.isValue;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.util.stream.Collectors.toList;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.contains;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static rx.Observable.just;
+
+// TODO it is still difficult to decouple + test the configured plugins + services
```

Comment:
Looks like you are having difficulties to test service loading. That is because the services are loaded by a separate static method which cannot be mocked.